tokio-executor/src/global.rs

```diff
@@ -150,6 +191,66 @@ pub fn spawn<T>(future: T)
         .unwrap()
 }
 
+/// Submits a future for execution on the default executor, returning a
+/// [`SpawnHandle`] that allows access to the result of the spawned future
+/// and can cancel it if it is no longer necessary.
+///
+/// Spawning the future behaves similarly to the [`spawn`] function, but with
+/// the addition of returning a `SpawnHandle`. A `SpawnHandle` is itself a
+/// future which will eventually complete with the value returned by the
+/// spawned future.
+///
+/// If the spawned future is no longer needed, the [`SpawnHandle::cancel`]
```

Comment:
IMO we want the reverse behavior. By default, when the handle is dropped, the spawned future is canceled. There should be a `forget` function that allows the task to continue running.

The [impl in futures 0.1](https://docs.rs/futures/0.1.24/futures/sync/oneshot/fn.spawn.html) is mostly correct, except that it panics when the task does not complete execution.